pkg/block/azure/walker.go

```diff
@@ -83,12 +84,17 @@ func (a *BlobWalker) Walk(ctx context.Context, storageURI *url.URL, op block.Wal
 			if op.After != "" && *blobInfo.Name <= op.After {
 				continue
 			}
+			// Skip folders
+			if *blobInfo.Properties.ContentLength == 0 && blobInfo.Properties.ContentMD5 == nil {
```

Comment:
In the past ContentMD5 was an optional field and it was possible to upload objects without calculating the MD5 sum.
This probably means that there's an edge case where a file with size 0 was uploaded with request not to calculate MD5 sum will be skipped on import.
We might want to investigate / find alternatives for this solution
https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.blob.blobrequestoptions.storeblobcontentmd5?view=azure-dotnet-legacy

Reply:
I'm not saying we should remove it - I'm suggesting it is not a sufficient validation.
Another question: why are we using the MD5 as ETag if object ETag exists?
Also would ETag exist for directory markers?

Reply:
The entry metadata we keep `checksum` the value is calculated as the content md5 of the object we upload.
The Etag value is `0x([0-9A-F]{15}` which is used for caching, but the value doesn't represent 1:1 the content. 
Specific the associate bug was importing two objects that holds the same Etag value.
Having the MD5 value as checksum will align the value if you upload the content through lakeFS (upload) or indirect using pre-signed URL and now also when you import the data.
My concern about directory marker is not because Azure keeps directory marker, it is because when you list the ADLS using blob API it will report a blob for a directory (not exactly a marker) and in this case we will import an entry with no checksum.
When you upload / create directory marker or any zero length content the checksum is `d41d8cd98f00b204e9800998ecf8427e`. So I expected that if they had a real object/blob behind the directory representation, it will also include this value as content MD5.